### ISSUE TITLE
IAM calls failing: "Attribute (_request_id) does not pass the type constraint because: Validation failed for 'Str'"

### ISSUE THREAD

**Issue by scampbell123** [pplu/aws-sdk-perl#324]
I have been querying the IAM service without issue for over a year now.  Since last night, about 90% of my calls now fail with this error:

Attribute (_request_id) does not pass the type constraint because: Validation failed for 'Str' with value [ "blah", "blah ]

I can re-run it, and it will work, or fail.  It almost seems like it's an AWS API issue and I randomly seem to be querying a bad AWS API server, but not sure.

Here's my code:
my $iam = $paws->service('IAM');
eval {$res = $iam->ListUsers()};
print $@->{'message'};

Anyone else seeing this?  I am doing it from the us-east-1 region.  

Out of curiosity, I ran a ListUsers without PAWS, just using the AWS CLI, and it works 100% of the time: "aws iam list-users --output text".

**Comment by Prajithp** [comment]
The issue happened to me as well. Looks like issue with one of the aws endpoint server as I could see duplicate x-amzn-requestid header sets in the response for some requests. This is happening only once in 2 or 3 requests.
```bash
 dragonfly ~/bin python3 aws_client.py 
{'content-length': '31012', 'x-amzn-requestid': '61afafa8-61fb-11e9-b852-5be9afe00d72', 'vary': 'accept-encoding', 'date': 'Thu, 18 Apr 2019 16:59:50 GMT', 'content-type': 'text/xml'}
 dragonfly ~/bin python3 aws_client.py 
{'date': 'Thu, 18 Apr 2019 16:59:54 GMT', 'x-amzn-requestid': 'cf591abe-b666-4900-ae78-1997f8a1dcf9, cf591abe-b666-4900-ae78-1997f8a1dcf9', 'content-length': '31012', 'content-type': 'text/xml', 'vary': 'accept-encoding,Accept-Encoding'}
``
as a quick fix, just patch the Paws/Net/XMLResponse.pm with 
$request_id = ('ARRAY' eq ref $request_id) ? $request_id->[0] : $request_id;

**Comment by scampbell123** [comment]
Thanks.  I assumed it must be an AWS issue, but was surprised that their CLI didn't have the problem.  But maybe their CLI talks to different servers than the Paws calls?

**Comment by Prajithp** [comment]
@scampbell123  No,  aws cli uses different method to capture request_id  https://github.com/boto/botocore/blob/c06a288098f348aa8c0fbabf640015e817dd0ca5/botocore/parsers.py#L517

**Comment by pplu** [comment]
Sorry for not noticing that this basically breaks all IAM calls. Initially I thought that it was only error responses were affected.

I'm patching/testing this behavior to let Paws accept more than one Request ID in the headers, and will release a new Paws version immediately, since this seems to be affecting all IAM calls.

If @scampbell123 and @Prajithp can open support cases to AWS to ask the IAM team to not send duplicate headers, it would be nice, since this seems to be some type of strange behaviour from their side.

**Comment by pplu** [comment]
Just to sum up the bug, and give guidance to people: The IAM service has started returning duplicate request id headers (which come up as an array in HTTP objects). Since the response objects can only hold one Request ID we get this error.

Paws has been patched and 0.40 is being cut to solve this problem, and will soon be uploaded to CPAN.

Versions prior to 0.40 of Paws have been rendered unable to call IAM (since these responses cannot be processed), unless AWS stops sending these duplicate headers.

IF you cannot update Paws, you can manually patch the XMLResponse and JSONResponse classes, based on https://github.com/pplu/aws-sdk-perl/pull/329/commits/3f2078f72dc431949d2eda5c181d553bf18f7d9e.

**Comment by pplu** [comment]
Paws 0.40 has just been uploaded to CPAN with a fix for this issue
